src/playbackwindow.cpp

```diff
@@ -359,10 +380,28 @@ void playbackWindow::checkForSaveDone()
 		ui->lblFrameRate->setText(tmp);
 		setControlEnable(false);
 
+		struct statvfs statvfsBuf;
+		statvfs(camera->recorder->fileDirectory, &statvfsBuf);
+		qDebug("Free space: %llu  (%lu * %lu)", statvfsBuf.f_bsize * (uint64_t)statvfsBuf.f_bfree, statvfsBuf.f_bsize, statvfsBuf.f_bfree);
+		
 		/* Prevent the user from pressing the abort/save button just after the last frame,
 		 * as that can make the camera try to save a 2nd video too soon, crashing the camapp.*/
 		if(camera->playFrame >= markOutFrame - 25)
 			ui->cmdSave->setEnabled(false);
+		
+		/*Abort the save if insufficient free space,
+		but not if the save has already been aborted,
+		or if the save button is not enabled(unsafe to abort at that time)(except if save mode is RAW)*/
+		QSettings appSettings;
+		bool insufficientFreeSpace_current = (MIN_FREE_SPACE > statvfsBuf.f_bsize * (uint64_t)statvfsBuf.f_bfree);
+		if(insufficientFreeSpace_current &&
+		   !saveAborted &&
+				(ui->cmdSave->isEnabled() ||
+				appSettings.value("recorder/saveFormat", 0).toUInt() != SAVE_MODE_H264)
```

Comment:
Reading the appSetting should be put into a function like getSaveMode() or something. Would minimize the complexity of this expression and keep all the appSettings stuff in one place.